lib/linters.py

```diff
@@ -0,0 +1,627 @@
+    # For includes, which don't have a header, assume the lowest term
+    # length, as it could be included in a policy with any target
+    if not header:
+      max_len = min(self.config['MAX_TERM_LENS'].values())
+      if len(term.name) > max_len:
+        msg = ("Term %s is longer than %d characters" %
+             (term.name, max_len))
+        self.add(Severity.WARNING, msg, lineno=term.name.lineno)
+    # for policy files, which have headers, only check for the appropriate
+    # target
+    else:
+      header_targets = {x.platform for x in header.target}
+      for target, max_len in self.config['MAX_TERM_LENS'].items():
+        if target in header_targets and len(term.name) > max_len:
+          msg = ("Term %s is longer than %d characters" %
+               (term.name, max_len))
+          self.add(Severity.WARNING, msg, lineno=term.name.lineno)
+
+  def check_network(self, network):
+    max_len = self.config['MAX_NETWORK_LEN']
+    if len(network.name) > max_len:
+      msg = 'Network name %s is more than %d characters' % (network.name,
+                                  max_len)
+      self.add(Severity.WARNING, msg, network.name.lineno)
+
+  def check_service(self, service):
+    max_len = self.config['MAX_SERVICE_LEN']
+    if len(service.name) > max_len:
+      msg = 'Service name %s is more than %d characters' % (service.name,
+                                  max_len)
+      self.add(Severity.WARNING, msg, service.name.lineno)
+
+
+@register_linter
+class ExpiredTermEnforcer(BaseLintRule):
+
+  """ Notifies about terms that are expired or expiring soon. """
+
+  DEFAULTS = {
+    'WARN_DAYS': 14,
+    'ADVICE_DAYS': 30,
+    'ERROR_DAYS': 0,
+  }
+
+  def check_term(self, policy, header, term):
+    if term.expiration:
+      sev = None
+      days = (term.expiration - datetime.date.today()).days
+      if days < self.config['ERROR_DAYS']:
+        sev = Severity.ERROR
+      elif days < self.config['WARN_DAYS']:
+        sev = Severity.WARNING
+      elif days < self.config['ADVICE_DAYS']:
+        sev = Severity.ADVICE
+      if sev is not None:
+        if days < 0:
+          msg = 'term is expired!'
+        else:
+          msg = '%d days until term expiration' % days
+        self.add(sev, msg, lineno=term.name.lineno)
+
+
+@register_linter
+class EmptyNetworkOrPorts(BaseLintRule):
+
+  """ Ensures network and service objects have an element within them. """
+
+  def check_service(self, service):
+    if not len(service.items):
+      msg = 'Services must contain at least one element'
+      self.add(Severity.ERROR, msg, lineno=service.name.lineno)
+
+  def check_network(self, network):
+    if not len(network.items):
+      msg = 'Networks must contain at least one element'
+      self.add(Severity.ERROR, msg, lineno=network.name.lineno)
+
+
+@register_linter
+class CheckValidNetwork(BaseLintRule):
+
+  """ Ensures network IP addresses are accurate and in the proper format,
```

Comment:
Same problems with the above docstring. Please try to limit docstrings to a single line also. You can find examples that give further information in a paragraph after the first line.
A shorter line would be "Ensures IP addresses are correct and finds large IPv6 Networks."